core/src/test/java/org/infinispan/functional/FunctionalInMemoryTest.java

```diff
@@ -95,8 +96,8 @@ public void testExceptionPropagation(boolean isOwner, WriteMethod method) {
                (BiConsumer<WriteEntryView<String>, Void> & Serializable) (view, nil) -> {
                   throw new TestException();
                }, getClass());
-         fail("Should throw CacheException:TestException");
-      } catch (CacheException e) { // catches RemoteExceptions, too
+         fail("Should throw [CompletionException:]CacheException:TestException");
+      } catch (CacheException | CompletionException e) { // catches RemoteExceptions, too
```

Comment:
I'd be more comfortable knowing when a `CompletionException` is expected here instead of assuming that it's possible regardless of method and ownership.